components/raftstore/src/store/peer_storage.rs

```diff
@@ -1003,72 +1000,57 @@ where
         "region_id" => region_id,
     );
 
-    let msg = kv_snap
+    let apply_state: RaftApplyState = kv_snap
         .get_msg_cf(CF_RAFT, &keys::apply_state_key(region_id))
-        .map_err(into_other::<_, raft::Error>)?;
-    let apply_state: RaftApplyState = match msg {
-        None => {
-            return Err(storage_error(format!(
+        .and_then(|res| match res {
+            None => Err(box_err!(
```

Comment:
This is too implicit. What will the exact error be returned?

Reply:
I don't know, probably `Error::Other`. L1026 does the same way.

Reply:
refine it